### PR TITLE
azure Gradle plugin and host.json local.setting.json to match values in sample Azure Functions application

### PR DESCRIPTION
**Pull request by sdelamo**
see: https://plugins.gradle.org/plugin/com.microsoft.azure.azurefunctions
see: https://learn.microsoft.com/en-us/azure/azure-functions/functions-create-first-java-gradle
see: https://github.com/Azure-Samples/azure-functions-samples-java

### CODE REVIEW

File: starter-core/src/test/groovy/io/micronaut/starter/feature/function/azure/AzureFeatureValidatorSpec.groovy
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/function/azure/AzureFeatureValidatorSpec.groovy:1]: Validator was removed https://github.com/micronaut-projects/micronaut-starter/pull/1663

File: starter-azure-function/hosts.json
Comment (timyates) [inline starter-azure-function/hosts.json:1]: Shouldn't this be called `host.json` not `hosts.json`?

File: starter-core/src/main/resources/functions/azure/local.settings.json
Comment (timyates) [inline starter-core/src/main/resources/functions/azure/local.settings.json:1]: Do we need to also update `starter-azure-function/local.settings.json`?

**Comment by timyates** [comment]
The module `starter-azure-function` is only included in `settings.gradle` for java 8

Since https://azure.microsoft.com/en-gb/updates/java-11-support-for-azure-functions-is-now-generally-available/

Isn't Java 11 supported also?

Actually

https://azure.microsoft.com/en-gb/updates/generally-available-azure-functions-support-for-java-17/

Looks like Java 17 is also supported now?

**Comment by sdelamo** [comment]
> The module `starter-azure-function` is only included in `settings.gradle` for java 8
> 
> Since https://azure.microsoft.com/en-gb/updates/java-11-support-for-azure-functions-is-now-generally-available/
> 
> Isn't Java 11 supported also?
> 
> Actually
> 
> https://azure.microsoft.com/en-gb/updates/generally-available-azure-functions-support-for-java-17/
> 
> Looks like Java 17 is also supported now?

Yes, they support now 17

File: starter-azure-function/hosts.json
Comment (sdelamo) [reply starter-azure-function/hosts.json:1]: Probably, it should be `host.json`.

Comment (sdelamo) [reply starter-azure-function/hosts.json:1]: done via https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/588933906ea26589b2a81267957599091bf4c1c4

File: starter-core/src/main/resources/functions/azure/local.settings.json
Comment (sdelamo) [reply starter-core/src/main/resources/functions/azure/local.settings.json:1]: done via https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/588933906ea26589b2a81267957599091bf4c1c4

**Comment by sdelamo** [comment]
I have [included azure-function 11 and 17](https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/e1ab20135a8ba4a30571785c63b53ab4f8f19ffd)

**Review by timyates — APPROVED** [review]

**Review by timyates — APPROVED** [review]
[I think we need to update this to include java 17](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L117-L126)

Also, we seem to specify [windows for maven builds](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L109), but linux for [gradle builds](https://github.com/micronaut-projects/micronaut-starter/blob/e1ab20135a8ba4a30571785c63b53ab4f8f19ffd/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/template/azurefunctions.rocker.raw#L13)

Also, Maven builds have the [java version specified](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L110), but we don't do the same for Java

**Comment by sdelamo** [comment]
> [I think we need to update this to include java 17](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L117-L126)

> Also, Maven builds have the [java version specified](https://github.com/micronaut-projects/micronaut-starter/blob/abaa948e6c46fd17fa2cd74b95a641f993ea81a7/starter-core/src/main/java/io/micronaut/starter/feature/function/azure/AbstractAzureFunction.java#L110), but we don't do the same for Java

Done for both https://github.com/micronaut-projects/micronaut-starter/pull/1666/commits/dc493ea61b047649e8209b38d0b23753afce6d38

**Pull request merged**